### PR TITLE
meson init --build should call meson setup and meson compile

### LINKED ISSUE #11235: meson init --name xyz --build doesn't work on Windows - doesn't follow its own advice
Pardon the screenshot-as-a-bug-report, I was working on this in a VM as I don't generally use Windows.

image

It seems when passing the --build flag to meson init, it runs ninja directly rather than the meson compile wrapper, which leads to a failed build as it does not automatically activate the visual studio environment (which is an *excellent* feature, by the way!).

Running meson compile -C build works fine, as shown

This is on Windows 10, meson 0.64.1, installed with the MSI installer. Native build.

### PR DESCRIPTION
**Pull request by bruchar1**
Fixes #11235

### CODE REVIEW

**Comment by bruchar1** [comment]
> This is the only useful thing `compile` is doing (and the fact you ever need to do it is bad but that's VS architectural decisions for you).

Isn't `meson compile` allowing to be agnostic on the used backend?

**Comment by eli-schwartz** [comment]
`meson init` never invokes `meson setup`  with the `--backend vs` option, I think. Which is the only case that the backend would be not-ninja.

File: mesonbuild/minit.py [hunk 1]
```diff
@@ -22,7 +22,9 @@
 import os
 import re
 from glob import glob
-from mesonbuild import mesonlib
+from . import mlog
+from .mesonlib import setup_vsenv
```
Comment (eli-schwartz) [inline mesonbuild/minit.py:26]: We can just use `mesonlib.setup_vsenv` instead of importing this import twice, probably.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
Can you please squash both commits into one? :)

**Review by eli-schwartz — APPROVED** [review]

File: mesonbuild/minit.py [hunk 2]
```diff
@@ -22,7 +22,8 @@
 import os
 import re
 from glob import glob
-from mesonbuild import mesonlib
+from . import mlog
+from mesonbuild import build, mesonlib
```
Comment (eli-schwartz) [inline mesonbuild/minit.py:26]:
```suggestion
from mesonbuild import build, mesonlib, mlog
```

**Pull request merged**